src/CommunityToolkit.Mvvm.Fixers/FieldReferenceForObservablePropertyFieldFixer.cs

```diff
@@ -15,57 +13,74 @@
 
-        string? propertyName = diagnostic.Properties[FieldReferenceForObservablePropertyFieldAnalyzer.PropertyNameKey];
-        string? fieldName = diagnostic.Properties[FieldReferenceForObservablePropertyFieldAnalyzer.FieldNameKey];
-
-        if (propertyName == null || fieldName == null)
+        // Retrieve the properties passed by the analyzer
+        if (diagnostic.Properties[FieldReferenceForObservablePropertyFieldAnalyzer.PropertyNameKey] is not string propertyName ||
+            diagnostic.Properties[FieldReferenceForObservablePropertyFieldAnalyzer.FieldNameKey] is not string fieldName)
         {
             return;
         }
 
         SyntaxNode? root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
-        Debug.Assert(root != null);
-
-        IdentifierNameSyntax fieldReference = root!.FindNode(diagnosticSpan).DescendantNodesAndSelf().OfType<IdentifierNameSyntax>().FirstOrDefault(i => i.ToString() == fieldName);
 
-        if (fieldReference == null)
+        foreach (SyntaxNode syntaxNode in root!.FindNode(diagnosticSpan).DescendantNodesAndSelf())
         {
-            return;
-        }
-
-        context.RegisterCodeFix(
-            CodeAction.Create(
-                title: "Reference property",
-                createChangedDocument: c => UpdateReference(context.Document, fieldReference, propertyName, c),
-                equivalenceKey: "Reference property"),
-            diagnostic);
+            // Find the first descendant node from the source of the diagnostic that is an identifier with the target name
+            if (syntaxNode is IdentifierNameSyntax { Identifier.Text: string identifierName } identifierNameSyntax &&
+                identifierName == fieldName)
+            {
+                // Register the code fix to update the field reference to use the generated property instead
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        title: "Reference property",
+                        createChangedDocument: token => UpdateReference(context.Document, identifierNameSyntax, propertyName, token),
+                        equivalenceKey: "Reference property"),
+                    diagnostic);
 
+                return;
+            }
+        }
     }
 
+    /// <summary>
+    /// Applies the code fix to a target identifier and returns an updated document.
+    /// </summary>
+    /// <param name="document">The original document being fixed.</param>
+    /// <param name="fieldReference">The <see cref="IdentifierNameSyntax"/> corresponding to the field reference to update.</param>
+    /// <param name="propertyName">The name of the generated property.</param>
+    /// <param name="cancellationToken">The cancellation token for the operation.</param>
+    /// <returns>An updated document with the applied code fix, and <paramref name="fieldReference"/> being replaced with a property reference.</returns>
```

Comment:
There are more comments explaining what than there are lines of code showing what 🙂. These frankly feel useless. I'm not saying you can't include them... but these aren't expanding anything. They're not explaining any whys, just restating the what of the private method that can be just as easily seen from the code.

Reply:
Resolving this as we chatted on Discord, though we disagreed on the conclusion ahah 😄